claimManagement/src/main/java/org/openimis/imisclaims/ToRestApi.java

```diff
@@ -64,6 +64,14 @@ public HttpResponse postToRestApi(Object object, String functionName, boolean ad
             HttpResponse response = httpClient.execute(httpPost);
             int responseCode = response.getStatusLine().getStatusCode();
             Log.i("HTTP_POST", uri + functionName + " - " + responseCode);
+            if (responseCode >= 400) {
+                Log.e("HTTP_POST", object.toString());
+                HttpEntity entity = response.getEntity();
+                if (entity != null) {
+                    String errorPayload = EntityUtils.toString(entity);
+                    Log.e("HTTP_POST", "error payload" + errorPayload);
+                }
+            }
```

Comment:
This code will break any error handling requiring the response content, as it is consumed upon read. I added similar code to the `getContent` method so it can be logged and used for error handling.